lib/mix/lib/mix/tasks/compile.app.ex

```diff
@@ -18,13 +18,16 @@ defmodule Mix.Tasks.Compile.App do
   function in your `mix.exs` with the following options:
 
     * `:applications` - all applications your application depends
-      on at runtime. For example, if your application depends on
-      Erlang's `:crypto`, it needs to be added to this list. Most
-      of your dependencies must be added as well (unless they're
-      a development or test dependency). Mix and other tools use this
-      list in order to properly boot your application dependencies
+      on at runtime. By default, this list is automatically inflected
+      from your dependencies. Any extra Erlang/Elixir dependency
+      must be specified in `:extra_applications`. Mix and other tools
+      use the application list in order to start your dependencies
       before starting the application itself.
 
+    * `:extra_applications` - a list of Erlang/Elixir applications
+      that you want started before your application. For example,
+      Elixir's `:logger` or Erlang's `:crypto`.
```

Comment:
On this list there is a final period on the items. I am just keeping it consistent with the surrounding lists (although Elixir is not consistent).